src/main/scala/wasm/StagedMiniWasm.scala

```diff
@@ -0,0 +1,622 @@
+package gensym.wasm.miniwasm
+
+import scala.collection.mutable.{ArrayBuffer, HashMap}
+
+import lms.core.stub.Adapter
+import lms.core.virtualize
+import lms.macros.SourceContext
+import lms.core.stub.{Base, ScalaGenBase}
+import lms.core.Backend._
+import lms.core.Backend.{Block => LMSBlock}
+
+import gensym.wasm.ast._
+import gensym.wasm.ast.{Const => WasmConst, Block => WasmBlock}
+import gensym.lmsx.{SAIDriver, StringOps, SAIOps, SAICodeGenBase}
+
+@virtualize
+trait StagedWasmEvaluator extends SAIOps {
+  def module: ModuleInstance
+  // NOTE: we don't need the following statements anymore, but where are they initialized?
```

Comment:
See https://github.com/TiarkRompf/lms-clean/blob/master/src/main/scala/lms/core/stub.scala#L62 and https://github.com/TiarkRompf/lms-clean/blob/master/src/main/scala/lms/core/frontend.scala#L107